### PR TITLE
chitter challenge - it was done last night as well

### PR DESCRIPTION
**Pull request by RizAli**

### CODE REVIEW

File: lib/server.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
+require 'data_mapper'
+require 'sinatra'
+require_relative 'user'
+require_relative 'peep'
+require './lib/helpers/application'
+
+env = ENV['RACK_ENV'] || 'development'
+
+# DataMapper::Logger.new($stdout, :debug)
```
Comment (tansaku) [inline lib/server.rb]: avoid leaving comments in like this - commit them and then you'll have them in your github history - you don't want dead code in things you submit to github

File: lib/server.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+DataMapper.setup(:default, "postgres://localhost/chitter_challenge_#{env}")
+
+require './lib/user'
+DataMapper.finalize
+# DataMapper.auto_migrate!
```
Comment (tansaku) [inline lib/server.rb]: avoid leaving comments in like this - commit them and then you'll have them in your github history - you don't want dead code in things you submit to github

File: lib/server.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+DataMapper.auto_upgrade!
+
+enable :sessions
+set :session_secret, 'super secret'
+
+
+get '/' do
```
Comment (tansaku) [inline lib/server.rb]: You should prefer to have this inside a class that extends Sinatra::Application

File: lib/helpers/application.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+helpers do
```
Comment (tansaku) [inline lib/helpers/application.rb]: nice to see this in a separate file

File: Gemfile [hunk 5]
```diff
@@ -0,0 +1,16 @@
+source 'https://rubygems.org'
+
+gem 'sinatra'
+gem 'data_mapper'
+gem 'dm-postgres-adapter'
+gem 'capybara'
```
Comment (tansaku) [inline Gemfile]: should be in test group

File: lib/helpers/application.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+
+  def current_user
+    @current_user ||=User.get(session[:user_id]) if session[:user_id]
```
Comment (tansaku) [inline lib/helpers/application.rb]: please ensure space after '||='

**Comment by tansaku** [comment]
All the tests pass which is good. I can post peeps, but I don't see any signin/signup functionality on the main page.  I was able to access that via users/new, but as you probably know you should be updating your layout.erb to include the rest of the necessary functionality.

This is a good start, but you'll need to follow through if you want to get a badge.

:finished:

**Comment by spike01** [comment]
:construction:

**Pull request closed without merging**